### PR TITLE
Swagger: Re-generate the enterprise specification if enterprise is cloned

### PR DESCRIPTION
**Pull request by papagian**
**What is this feature?**

https://github.com/grafana/grafana-backend-platform-squad/issues/41

**Why do we need this feature?**

- makes sure that the swagger OSS and enterprise specifications are kept synced (by building the server with enterprise enabled)
- silences logs during generation
- prints the diff in case of inconsistencies to help authors understand the source of the step failure [example](https://drone.grafana.net/grafana/grafana/158436/9/3)

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/api/folder.go [hunk 1]
```diff
@@ -31,7 +31,7 @@ const REDACTED = "redacted"
 //
 // Get all folders.
 //
-// Returns all folders that the authenticated user has permission to view.
+// It returns all folders that the authenticated user has permission to view.
```
Comment (papagian) [inline pkg/api/folder.go:34]: this change is only for triggering the `swagger-gen` step (triggered only if there are backend changes)

File: .drone.yml [hunk 2]
```diff
@@ -1234,11 +1234,15 @@ steps:
     | jq .head.repo.fork)
   - if [ "$is_fork" != false ]; then return 1; fi
   - git clone "https://$${GITHUB_TOKEN}@github.com/grafana/grafana-enterprise.git"
-    grafana-enterprise
-  - cd grafana-enterprise
+    ../grafana-enterprise
+  - cd ../grafana-enterprise
```
Comment (aangelisc) [inline .drone.yml:1238]: Maybe a stupid question - aren't this line and the line above equivalent?

**Review by aangelisc — APPROVED** [review]
Changes LGTM 😊

File: scripts/drone/pipelines/swagger_gen.star [hunk 3]
```diff
@@ -33,7 +33,7 @@ def swagger_gen_step(ver_mode):
             "apk add --update git make",
             "make swagger-clean && make openapi3-gen",
             "for f in public/api-merged.json public/openapi3.json; do git add $f; done",
-            'if [ -z "$(git diff --name-only --cached)" ]; then echo "Everything seems up to date!"; else echo "Please ensure the branch is up-to-date, then regenerate the specification by running make swagger-clean && make openapi3-gen" && return 1; fi',
+            'if [ -z "$(git diff --name-only --cached)" ]; then echo "Everything seems up to date!"; else git diff --cached && echo "Please ensure the branch is up-to-date, then regenerate the specification by running make swagger-clean && make openapi3-gen" && return 1; fi',
```
Comment (alexweav) [inline scripts/drone/pipelines/swagger_gen.star:36]: There was another issue later on today where the swagger docs did drift in `main` which stalled most builds. (Basically, a change managed to escape the CI step)

Thread:
https://raintank-corp.slack.com/archives/C04TFFPTSNS/p1709755662500559
PR: https://github.com/grafana/grafana/pull/84037

In the process of fixing that, I brought in this one line of change to print the diff - we needed it to make debugging easier.

I pinged you in the commit that added this: https://github.com/grafana/grafana/pull/84037/commits/2c6037adc432766fffe0deb75b9b377b39b60b0c to hopefully attribute it correctly as it was pilfered directly from this PR.

So, this one tiny piece is already merged. I think all it needs is a .drone.yml regeneration.

**Review by alexweav — APPROVED** [review]

File: .drone.yml (hunk 2 above)
Comment (papagian) [reply .drone.yml:1238]: I think this is because I switched to using `enterprise_setup_step` instead of `clone_enterprise_step_pr`

**Pull request merged**